### ISSUE TITLE
DRY and Select2 filters.js

### ISSUE THREAD

**Issue by a2xchip** [akeneo/pim-community-dev#4382]
**I'm reporting a Bug** && **I'm asking a Question**

There are two almost same js files: 

* https://github.com/a2xchip/pim-community-dev/blob/issue4356/src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-choice-filter.js
* and https://github.com/a2xchip/pim-community-dev/blob/issue4356/src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-rest-choice-filter.js

They both are using the same template: ```pim/template/datagrid/filter/select2-choice-filter```

And all difference I found is in that format they are receiving data from server:

* https://github.com/a2xchip/pim-community-dev/blob/issue4356/src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-choice-filter.js#L98

* https://github.com/a2xchip/pim-community-dev/blob/issue4356/src/Pim/Bundle/DataGridBundle/Resources/public/js/datafilter/filter/select2-rest-choice-filter.js#L90 

### Task/Question

**Will not it be better to remove one of them (preferably first), to comply DRY principals and standardize server side formating for that filter types using first ```select2-filter.js?```**

**Comment by phaseinducer** [comment]
Hi Alexander!

Indeed the "select2-choice-filter.js" is now legacy and shouldn't be used anymore.
After you point it out, I did a little research and it seems that it's now used only for filtering on select attributes (simple and multi). The response is handled by [this action](https://github.com/akeneo/pim-community-dev/blob/master/src/Pim/Bundle/UIBundle/Controller/AjaxOptionController.php#L41) which is also used for other selects than grid filters.
It became pretty unmaintainable over time so we need to work on a cleaner solution, it has been added to our technical backlog.

Thanks for the feedback ; )
Yohan

**Comment by a2xchip** [comment]
@phaseinducer How to rate your ```easy pick``` label removing? I want to place finger up there ))

**Comment by pierallard** [comment]
@a2xchip "Easy pick" was removed because after discussion, this is really not an easy pick, just to check everywhere that nothing is broken after merge these behaviors.

**Comment by a2xchip** [comment]
@pierallard Yeah, needed changes may be very dangerous without deep testing and checking.
